xdsl/dialects/vector.py

```diff
@@ -132,6 +132,10 @@ def get(
 
 @irdl_op_definition
 class BroadcastOp(IRDLOperation):
+    """
+    See external [documentation](https://mlir.llvm.org/docs/Dialects/Vector/#vectorbroadcast-vectorbroadcastop).
+    """
+
     name = "vector.broadcast"
     source = operand_def()
```

Comment:
Could we not do something like:
```
T: ClassVar = VarConstraint("T", AnyVectorElement)

...
source = operand_def(T | VectorType[T])
```
?

Reply:
The problem is that there isn't really a constraint on vector elements, so we can't prove that these are disjoint...

Reply:
I thought vector elements were a fairly small set of types

Reply:
In MLIR there's a `VectorElementTypeInterface`, we don't have any constraints, in either case it seems like it's an open thing, which we currently don't support in AnyOf.

Reply:
I think it's worth chatting about but I wouldn't rush into reverting it.